addons/sale/models/sale.py

```diff
@@ -511,22 +511,24 @@ def create(self, vals):
         return result
 
     def _compute_field_value(self, field):
+        if field.name == 'invoice_status':
+            filtered_self = self.filtered(lambda so: so._origin.invoice_status != 'upselling')
         super()._compute_field_value(field)
         if field.name != 'invoice_status' or self.env.context.get('mail_activity_automation_skip'):
             return
 
-        filtered_self = self.filtered(lambda so: so.user_id and so.invoice_status == 'upselling')
+        filtered_self = filtered_self.filtered(lambda so: so.user_id and so.invoice_status == 'upselling')
```

Comment:
I think that filtering directly on the `so.user_id` in the first filter would increase performance don't you think or am I missing something that would prevent doing it 🙂?